services/web-app/components/storybook-demo/storybook-demo.js

```diff
@@ -0,0 +1,125 @@
+/*
+ * Copyright IBM Corp. 2022, 2022
+ *
+ * This source code is licensed under the Apache-2.0 license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+import { Column, Dropdown, Grid } from '@carbon/react'
+import clsx from 'clsx'
+import PropTypes from 'prop-types'
+import { useState } from 'react'
+
+import * as styles from './storybook-demo.module.scss'
+
+const StorybookDemo = ({ height, themeSelector, size, url, variants }) => {
+  const columnSize = size === 'large' ? 12 : 8
+
+  const demoClassNames = clsx(styles['storybook-demo'], styles[`${height}`], {
+    [styles.large]: size === 'large'
+  })
+
+  const themeItems = [
+    {
+      id: 'white',
+      label: 'White',
+      src: 'white'
+    },
+    {
+      id: 'g10',
+      label: 'Gray 10',
+      src: 'g10'
+    },
+    {
+      id: 'g90',
+      label: 'Gray 90',
+      src: 'g90'
+    },
+    {
+      id: 'g100',
+      label: 'Gray 100',
+      src: 'g100'
+    }
+  ]
+
+  const [theme, setTheme] = useState('')
+  const onThemeChange = (item) => {
+    setTheme(item.selectedItem.src)
+  }
+
+  const initialSetVariant = variants[0].variant
+  const initialSelectedVariant = variants[0].label
+  const initialSelectedTheme = themeItems[0]
+
+  const [variant, setVariant] = useState(initialSetVariant)
+
+  const iframeUrl = url + '/iframe.html?id=' + variant + '&globals=theme:' + theme
+
+  const onVariantChange = (item) => {
+    setVariant(item.selectedItem.variant)
+  }
+
+  return (
+    <>
+      <Grid condensed>
+        {themeSelector && (
+          <Column sm={2} md={4}>
+            <Dropdown
+              titleText="Theme selector"
+              label="theme"
+              items={themeItems}
+              onChange={onThemeChange}
+              initialSelectedItem={initialSelectedTheme}
+            />
+          </Column>
+        )}
+        <Column sm={2} md={4}>
+          <Dropdown
+            titleText="Variant selector"
+            label="variant"
+            items={variants}
+            initialSelectedItem={initialSelectedVariant}
+            onChange={onVariantChange}
+          />
+        </Column>
```

Comment:
I got the impression from the redlines that the variant selector was optional, I tried not sending the variants but the component errored out, are we covering for that scenario? 
<img width="807" alt="image" src="https://user-images.githubusercontent.com/40550942/181620052-9d5381dc-bd97-4892-bef3-b55dcf12c35b.png">

Reply:
Yea, was thinking about this one too. How would we want them to input that into the mdx if they only have one story. 

Would we want something like this
```
<StorybookDemo
  themeSelector={true}
  size="small"
  height="tall"
  url="https://react.carbondesignsystem.com/iframe.html?args=&id=components-button--default"
/>
```

or this, and if there is only one variant, then no switcher shows up?
```

<StorybookDemo
  themeSelector={true}
  size="small"
  height="tall"
  url="https://react.carbondesignsystem.com"
  variants={[
    {
      label: 'Button',
      variant: 'components-button--default'
    }
  ]}
/>
```

I'm leaning toward option 2.

Reply:
we could support both? 👀